wasm/wasm_source/src/vp_masp.rs

```diff
@@ -70,9 +137,50 @@ fn validate_tx(
         }
 
         // Handle unshielding/transparent output
+        // The following boundary conditions must be satisfied
+        // 1. Zero transparent inupt
+        // 2. One transparent output
+        // 3. Asset type must be properly derived
+        // 4. Value from the output must be the same as the containing transfer
+        // 5. Public key must be the hash of the target
         if transfer.target != masp() {
```

Comment:
Where do we check for the following condition?: If the target address is the MASP validity predicate, then no transparent outputs are permitted in the shielded transaction.

It seems to me that if the source and target are MASP, then both the conditional on line 105 and line 146 are skipped. This would allow for these sorts of transactions to have unlimited transparent inputs and outputs.